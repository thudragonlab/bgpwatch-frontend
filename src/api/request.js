import axios from 'axios'
import {
  Message,
  MessageBox
} from 'element-ui'
// import { getToken } from '@/utils/auth'
// import { API_BASE_URL } from '@/utils/url_config'

// create an axios instance
let oldCrrentView = localStorage.getItem('currentView')
const service = axios.create({
  // baseURL: '', // url = base url + request url
  // withCredentials: true, // send cookies when cross-domain requests
  timeout: 30000 * 2
  // * 20, // request timeout
})

// request interceptor
service.interceptors.request.use(
  config => {
    // do something before request is sent
    // console.log(config)
    if (config.useToken) {
      const userInfo = localStorage.getItem('userInfo')
      if (userInfo) {
        const token = JSON.parse(userInfo).token
        // console.log(JSON.parse(userInfo).token)
        // config.headers['content-type'] = 'application/x-www-form-urlencoded;charset=UTF-8';
        config.headers['Authorization'] = 'Bearer ' + token
      }
    }

    return config
  },
  // error => {
  //   // do something with request error
  //   console.log(error) // for debug
  //   return Promise.reject(error)
  // }
)

// response interceptor
service.interceptors.response.use(
  /**
   * If you want to get http information such as headers or status
   * Please return  response => response
   */

  /**
   * Determine the request status by custom code
   * Here is just an example
   * You can also judge the status by HTTP Status Code
   */
  response => {
    const {
      data,
      headers
    } = response
    if (headers['content-type'] === "text/csv; charset=utf-8") {
      const fileName = headers['content-disposition'].replace(/\w+; filename=(.*)/, '$1')
      console.log(fileName)
      // 此处当返回json文件时需要先对data进行JSON.stringify处理，其他类型文件不用做处理
    } else {
      const data = response.data
      if(data.lastUpdateTimeStamp && data.lastUpdateTimeStamp > 0){
        const lastUpdateTimeStamp = data.lastUpdateTimeStamp
        const nowCrrentView = localStorage.getItem('currentView')
        if(oldCrrentView !== nowCrrentView){
          localStorage.setItem('lastUpdateTimestamp',lastUpdateTimeStamp)
          oldCrrentView = nowCrrentView
        }else{
          const oldLastUpdateTimestamp = localStorage.getItem('lastUpdateTimestamp')
          if(!oldLastUpdateTimestamp){
            localStorage.setItem('lastUpdateTimestamp',lastUpdateTimeStamp)  
          }else{
            const newUpdateTimestamp = Math.max(Number(lastUpdateTimeStamp),Number(oldLastUpdateTimestamp))
            localStorage.setItem('lastUpdateTimestamp',newUpdateTimestamp)
          }
        }
        
      }
      
      if (!data.status) {
        MessageBox({
          message: data.statusCode || 'Error',
          type: 'error',
          duration: 5 * 1000
        })
        return Promise.reject(data)
      }
    }
    return data.data
   
  },
  error => {
    console.error(error.message) // for debug
    if (error.message && error.message.includes('aborted')) {
      return Promise.reject(error)
    }

    MessageBox({
      message: 'No data found.',
      type: 'error',
      duration: 1 * 1000
    })
    return Promise.reject(error)
  }
)

export default service