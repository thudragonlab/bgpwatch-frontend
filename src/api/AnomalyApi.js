import request from './request'
const MY_URL = '/myapi'
export function getEventlistByDate(st, et) {
    return request({
        methods: 'GET',
        url: `${MY_URL}/events?st=${st}&et=${et}&event_type=ALL`,
        timeout: 3 * 60 * 1000,
    })
}
export function getMoasEventlistByDate(st, et) {
    return request({
        methods: 'GET',
        url: `${MY_URL}/events?st=${st}&et=${et}&event_type=ALL_MOAS`,
        timeout: 3 * 60 * 1000,
    })
}


export function getEventDetail(detail_url) {
    // return new Promise((resolve,reject) => {
    //   return resolve(eventsDetail)
    // })
    return request({
        methods: 'GET',
        url: `${MY_URL}${detail_url}`
    })
}


