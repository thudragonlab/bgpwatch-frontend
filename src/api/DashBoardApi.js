import request from './request'
import axios from 'axios'
// const BASE_URL = 'http://10.110.62.34:5000/'
const BASE_URL = '/dbapi'
const BASE_URL2 = '/newapi'
const MY_URL = '/myapi'
const WZQ_API2 = '/wzqApi2'


export function getAsInfoByASN(asn) {
  return request({
    method: 'GET',
    url: `${BASE_URL}/api/dashboard/daily/asinfo?asn=${asn}`,

  })
}

export function getAsInfoByASN_myAPI(asn) {
  return request({
    method: 'GET',
    url: `${MY_URL}/asInfo?asn=${asn}`,

  })
}


export function getPeerPrefix(asn, {
  type,
  detailAsn
}) {
  return request({
    method: 'GET',
    url: `${BASE_URL}/api/dashboard/daily/peerprefix?asn=${asn}&${type}=${detailAsn}`,
  })
}



export function getAsRoute(asn, type) {
  return request({
    method: 'GET',
    url: `${MY_URL}/getAsRoute?asn=${asn}&type=${type}`,
  })
}

export function getPeerTypeDistribute(asn) {
  return request({
    method: 'GET',
    url: `${MY_URL}/getPeerTypeDistribute?asn=${asn}`,
  })
}


export function getAsInfoByList(asnList) {
  return request({
    method: 'POST',
    url: `${MY_URL}/getAsInfoByList`,
    data: asnList.map(as => parseInt(as))
  })
}

export function getRealTimeData() {
  return request({
    method: 'GET',
    url: `${MY_URL}/getRealTimeData`,
  })
}

export function getRealTimeDataInAWhile() {
  return request({
    method: 'GET',
    url: `${MY_URL}/getRealTimeDataInAWhile`,
  })
}

export function getASList(asn) {
  return request({
    method: 'GET',
    url: `${MY_URL}/getAsList?asn=${asn}`,
  })
}

// import test from '@/../public/static/temporaryData/1.json'
export function getAsPathByPrefix(prefix, asn,type) {
  // return new Promise((resolve,reject) => {
  //   resolve(test)
  // })
  return request({
    method: 'GET',
    url: `${MY_URL}/getAsPathByPrefix?prefix=${prefix}&asn=${asn}&type=${type}`,
  })
}


export function getRouteHistoryStats(asn, qtype = 'g') {
  return request({
    method: 'GET',
    url: `${MY_URL}/route_history_stats?asns=${asn.join(',')}&qtype=${qtype}`,
  })
}


export function getRoutingPath(asn, pfx, qtype = "from") {
  return request({
    method: 'GET',
    url: `${WZQ_API2}/api/routing_path?asn=${asn}&pfx=${pfx}&qtype=${qtype}`
  })
}


export function getPrefixList(asn, qstr = undefined) {
  return request({
    method: 'GET',
    url: `${MY_URL}/prefix_list?asn=${asn}${qstr ? `&qstr=${qstr}` :  ''}`
  })
}

export function getPrefixTotalByAsn(asn) {
  return request({
    method: 'GET',
    url: `${MY_URL}/getPrefixTotalByAsn?asn=${asn}`,
  })
}

export function getPeerList(asn,ipVersion) {
  return request({
    method: 'GET',
    url: `${MY_URL}/peer_list?asn=${asn}&ipversion=${ipVersion}`
  })
}


// http://10.110.62.37:5050/api/routing_path?asn=4538&pfx=59.66.196.0%2F24&qtype=to

export function getWhoisInfoByASN(asn){
  return request({
    method: 'GET',
    url: `${MY_URL}/get_whois_info_by_asn?asn=${asn}`
  })
}