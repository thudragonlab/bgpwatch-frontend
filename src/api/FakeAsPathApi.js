import request from './request'
const MY_URL = '/myapi'

export function getFakeAsPath(st, et) {
    return request({
        methods: 'GET',
        url: `${MY_URL}/getFakeAsPath?st=${st}&et=${et}`,
        timeout: 3 * 60 * 1000,
    })
}


export function getFakeAsPathRecordById(id){

    return request({
        methods: 'GET',
        url: `${MY_URL}/getFakeAsPathRecordById?id=${id}`,
        timeout: 3 * 60 * 1000,
    })

}