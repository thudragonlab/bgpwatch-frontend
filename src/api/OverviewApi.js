import request from './request'
import axios from 'axios'
// const BASE_URL = 'http://10.110.62.34:5000/'
const MY_URL = '/myapi'


export function getOverviewData(startTime, endTime,selectedOption,asn) {
    return request({
      methods: 'GET',
      url: `${MY_URL}/overview?start=${startTime}&end=${endTime}&cc=${selectedOption}&asn=${asn}`,
    })
  }


export function downloadCSV(startTime, endTime) {
    return request({
        methods: 'GET',
        url: `${MY_URL}/exportCSV?start=${startTime}&end=${endTime}`,
        responseType:'blob'
      })
}