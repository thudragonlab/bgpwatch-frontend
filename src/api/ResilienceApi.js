import request from './request'
const MY_URL = '/myapi'

export function getResilienceByccCode(cc) {
  return request({
    method: 'GET',
    url: `${MY_URL}/getResilienceByccCode?cc=${cc}`,
  })
}


export function getResilienceTestData(cc, asn) {
  return request({
    method: 'GET',
    url: `${MY_URL}/getTopologyData?cc=${cc}&asn=${asn}`,
  })
}


export function getResilienceTestDataInOptimize(cc, asn) {
  return request({
    method: 'GET',
    url: `${MY_URL}/getTopologyDataOpt?cc=${cc}&asn=${asn}`,
  })
}

export function getAsByCc(cc) {
  return request({
    method: 'GET',
    url: `${MY_URL}/getAsByCc?cc=${cc}`,
  })
}


export function getRtreeByAsn(cc, asn, condition) {
  return request({
    method: 'GET',
    url: `${MY_URL}/getRtreeByAsn?cc=${cc}&asn=${asn}&condition=${condition}`,
  })
}

