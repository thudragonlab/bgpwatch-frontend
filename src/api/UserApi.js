import request from './request'
import axios from 'axios'
// const BASE_URL = 'http://10.110.62.34:5000/'
const MY_URL = '/myapi'

export function Login(user_name,password) {
    const fd = new FormData()
    fd.append('user_name', user_name)
    fd.append('password', password)
  return request({
    method: 'POST',
    url: `${MY_URL}/login`,
    data: fd
  })
}


export function Register(token) {
    return request({
      method: 'GET',
      url: `${MY_URL}/register?token=${token}`,
    })
  }

  export function VerifyMail(user_name,password,email) {
    const fd = new FormData()
    fd.append('user_name', user_name)
    fd.append('password', password)
    fd.append('email', email)
    return request({
      method: 'POST',
      url: `${MY_URL}/verifyEmail`,
      data: fd
    })
  }


export function forgetPwd(userName) {
  return request({
    method: 'GET',
    url: `${MY_URL}/forget?userName=${userName}`,
  })
}

export function updatePwd(userName,pwd){
  return request({
    method: 'POST',
    url: `${MY_URL}/updatePwd`,
    data: {
      user:userName,
      pwd
    }
  })
}

export function addSource(source){
  return request({
    method: 'GET',
    url: `${MY_URL}/addSourceIp?source=${source}`,
    useToken: true
  })
}

export function delSource(source){
  return request({
    method: 'GET',
    url: `${MY_URL}/delSourceIp?source=${source}`,
    useToken: true
  })
}
export function getSourceIP(){
  return request({
    method: 'GET',
    url: `${MY_URL}/getSourceIP`,
    useToken: true
  })
}

