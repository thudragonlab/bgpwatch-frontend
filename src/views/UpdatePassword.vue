<template>
  <div style="height: 100%">
    <update-password-inner :submitFunc="Submit" :decodeFunc="decode" />
  </div>
</template>

<script>
import { updatePwd } from '@/api/UserApi'
export default {
  name: 'UpdatePassword',
  methods: {
    async Submit(userName, ruleForm) {
      const result = await updatePwd(userName, ruleForm.pass)
      this.$alert(result.message, '', {
        confirmButtonText: 'OK',
        callback: action => {
          this.$message('Please login again')
          this.$router.push('/')
        },
      })
    },
    decode(userkey) {
      const decode_str = Buffer.from(userkey, 'base64').toString()
      return decode_str.split('dragonLab_TSU').join('')
    },
  },
}
</script>

<style lang="scss" scoped>
.statistic-box {
  // background-color: #f0f0f0;
  height: 100%;
  background-image: linear-gradient(45deg, #efbdfa, #cbe3fd, #f9d9d9);
  display: flex;
  justify-content: center;
  align-items: center;
}

.header {
  font-size: var(--max-font-size);
  font-weight: bold;
}

.box-card {
  width: 580px;
}

.bnt {
  margin: 0;
  margin-top: 10px;
  width: 100%;
}
</style>
