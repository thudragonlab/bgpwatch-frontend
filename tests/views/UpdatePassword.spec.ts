import {
  Wrapper,
  shallowMount
} from '@vue/test-utils'
import UpdatePassword from "@/views/UpdatePassword.vue";
import { updatePwd } from '@/api/UserApi'
import Vue from "vue";


Vue.prototype.$message = jest.fn()
Vue.prototype.$alert = jest.fn((_,__,{callback}) => {
  callback()
})


jest.mock('@/api/UserApi', () => {
  return {
    updatePwd: jest.fn(() => Promise.resolve({'message': 'Success'}))
  }
})


describe('UpdatePassword.vue', () => {
  it('renders correctly',async () => {
    const $route = {
      query: {
        user: 'test-token'
      },
      
    }
    const push = jest.fn()
    const wrapper:Wrapper<any> = shallowMount(UpdatePassword, {
      mocks: {
        $route,
        $router:{
          push
        }
      }
    })

    expect(wrapper.html()).toMatchSnapshot();
    const user = wrapper.vm.decode('ZHJhZ29uTGFiX1RTVTE=')
    expect(user).toBe('1')
    await wrapper.vm.Submit(user,{pass:'1'})
    expect(updatePwd).toHaveBeenCalled()
    expect(wrapper.vm.$alert).toHaveBeenCalledWith('Success', '', {
      confirmButtonText: 'OK',
      callback: expect.any(Function),
    });
    expect(wrapper.vm.$message).toHaveBeenCalledWith('Please login again');
    expect(push).toHaveBeenCalledWith('/');

  });
  
})