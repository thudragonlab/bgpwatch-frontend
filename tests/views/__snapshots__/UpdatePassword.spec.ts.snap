// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdatePassword.vue renders correctly 1`] = `
<div style="height: 100%;">
  <update-password-inner submitfunc="function () { [native code] }" decodefunc="function () { [native code] }"></update-password-inner>
</div>
`;
